src/Rox/Core/Repositories/TargetGroupRepository.php

```diff
@@ -26,6 +26,10 @@ function setTargetGroups($targetGroups)
      */
     function getTargetGroup($id)
     {
+        if ($this->_targetGroups === null) {
+            throw new \Exception('TargetGroupRepository._targetGroups is not set.');
+        }
```

Comment:
Many invalid argument usages were changed from `Exception` to `TypeError`, which means Rox's try-catches weren't working as expected. You'd have to implement `Throwable` to also catch `TypeError`, but that runs the risk of catching more things than originally intended.